File: truth_set/go/check/function/getUserName.go
package function

type UserResult struct {
    Name  *string
    Error *string
}

func getUserName(userId int) UserResult {
    if userId == 1 {
        name := "Alice"
        return UserResult{Name: &name, Error: nil}
    }
    errMsg := "User not found."
    return UserResult{Name: nil, Error: &errMsg}
}

func main() {
    result := getUserName(2)
    userName := *result.Name
    println("User Name:", userName)
}
\ No newline at end of file

File: truth_set/javascript/check/function/getUserName.js
function getUserName(userId) {
    if (userId === 1) {
        return { name: "Alice", error: null };
    }
    else {
        return { name: null, error: "User not found." };
    }
}
var result = getUserName(2);
console.log("User Name:", result.name.toUpperCase());